e2e_test/ddl/secret.slt

```diff
@@ -40,6 +40,23 @@ create secret secret_1 with (
   backend = 'meta'
 ) as 'demo_secret';
 
+statement ok
+alter secret secret_1 with (
+  backend = 'meta'
```

Comment:
The `backend` option should not appear in the `alter secret` statement. To follow the convention of other `alter` commands, only the changed part should be provided. Here the `backend` option is obviously untouched, so users should not write it here.

But, furthermore, I think `backend` shouldn't be a per-secret option. In my mind it should be a global-wise config and can't be changed after cluster initialized.

Reply:
> But, furthermore, I think backend shouldn't be a per-secret option. In my mind it should be a global-wise config and can't be changed after cluster initialized.

It's a bit off-topic. We can create a new issue